File: frontend/src/services/caseService.js
import api from './api';

export const caseService = {
  getAllCases: async () => {
    const response = await api.get('/cases');
    return response.data;
  },

  searchCases: async (query) => {
    const response = await api.get(`/cases/search?query=${encodeURIComponent(query)}`);
    return response.data;
  },

  getCaseById: async (id) => {
    const response = await api.get(`/cases/${id}`);
    return response.data;
  },

  createCase: async (caseData) => {
    const response = await api.post('/cases', caseData);
    return response.data;
  },

  updateCase: async (id, caseData) => {
    const response = await api.put(`/cases/${id}`, caseData);
    return response.data;
  },

  deleteCase: async (id) => {
    const response = await api.delete(`/cases/${id}`);
    return response.data;
  }
};

File: backend/src/main/java/com/forensic/service/CaseService.java
package com.forensic.service;

import com.forensic.entity.CaseEntity;
import com.forensic.repository.CaseRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class CaseService {

    private final CaseRepository caseRepository;

    public CaseService(CaseRepository caseRepository) {
        this.caseRepository = caseRepository;
    }

    public List<CaseEntity> getAllCases() {
        return caseRepository.findAll();
    }

    public Optional<CaseEntity> getCaseById(Long id) {
        return caseRepository.findById(id);
    }

    public Optional<CaseEntity> getCaseByNumber(String caseNumber) {
        return caseRepository.findByCaseNumber(caseNumber);
    }

    public List<CaseEntity> searchCases(String query) {
        return caseRepository.findByTitleContainingIgnoreCaseOrDescriptionContainingIgnoreCaseOrCaseNumberContainingIgnoreCase(query, query, query);
    }

    public CaseEntity createCase(CaseEntity caseEntity) {
        return caseRepository.save(caseEntity);
    }

    public CaseEntity updateCase(Long id, CaseEntity updatedCase) {
        return caseRepository.findById(id)
                .map(existingCase -> {
                    existingCase.setTitle(updatedCase.getTitle());
                    existingCase.setDescription(updatedCase.getDescription());
                    existingCase.setStatus(updatedCase.getStatus());
                    existingCase.setAssignedTo(updatedCase.getAssignedTo());
                    return caseRepository.save(existingCase);
                })
                .orElse(null);
    }

    public void deleteCase(Long id) {
        caseRepository.deleteById(id);
    }
}
